core/src/apps/cardano/README.md

```diff
@@ -1,27 +1,26 @@
 # Cardano
 
-MAINTAINER = Gabriel Kerekeš <gabriel.kerekes@vacuumlabs.com>
+MAINTAINER = Rafael Korbaš <rafael.korbas@vacuumlabs.com>
 
 ORIGINAL AUTHOR = Juraj Muravský <juraj.muravsky@vacuumlabs.com>
 
-SHELLEY UPDATE AUTHOR = Gabriel Kerekeš <gabriel.kerekes@vacuumlabs.com>
-
 REVIEWER = Jan Matejek <jan.matejek@satoshilabs.com>, Tomas Susanka <tomas.susanka@satoshilabs.com>
 
 -----
 
 ## Useful links
 
 [Cardano documentation](https://docs.cardano.org/en/latest/) - official documentation.
+[Cardano developer documentation](https://developers.cardano.org/) - official developer documentation.
 [Delegation Design Spec](https://hydra.iohk.io/build/2006688/download/1/delegation_design_spec.pdf) - contains information about delegation (addresses, certificates, withdrawals, ...).
-[Shelley CDDL spec](https://github.com/input-output-hk/cardano-ledger-specs/blob/460ee17d22cacb3ac4d90536ebe90500a356a1c9/shelley/chain-and-ledger/shelley-spec-ledger-test/cddl-files/shelley.cddl).
+[Multi Asset CDDL spec](https://github.com/input-output-hk/cardano-ledger-specs/blob/097890495cbb0e8b62106bcd090a5721c3f4b36f/shelley-ma/shelley-ma-test/cddl-files/shelley-ma.cddl).
 [Byron address format](https://github.com/input-output-hk/cardano-wallet/wiki/About-Address-Format---Byron).
 [The Shelley 1852' purpose and staking path](https://github.com/input-output-hk/implementation-decisions/blob/e2d1bed5e617f0907bc5e12cf1c3f3302a4a7c42/text/1852-hd-chimeric.md).
 [cbor.me](http://cbor.me/) - very useful tool for CBOR inspection.
 
 ## Important notes
 
-Unfortunately we are aware of the fact that currently at most ~14 inputs are supported per transaction. We suspect this is due to the memory heavy CBOR implementation. This should be fixed in the near future.
+Unfortunately we are aware of the fact that currently at most ~14 inputs are supported per transaction. To resolve this, the cardano app will have to be rewritten to support transaction streaming.
```

Comment:
wrt this, the CBOR implementation is massively inefficient because the slice operation makes a copy of the underlying bytes (plus creating a new object).

suggestion for refactor: use the `trezor.utils.BufferReader` class for huge efficiency gains :)

Reply:
Decoding is not a concern here, it's about cbor-encoding the signed transaction body so BufferWriter could make sense I guess. We indeed considered something similar but our initial sanity-check consisting of allocating an empty buffer before the cbor serialization happened didn't result in much more than was able to go through the cbor serialization. The improvement, would probably be several tens of percent at most on top of those 2-2.5 kB but the max tx size in Cardano is 16kB.

So cbor doesn't seem to be the real culprit and the gains from such refactor probably wouldn't be that spectacular.

Reply:
i'm not sure that SO question applies to micropython. But I don't know off the top of my head.

I would honestly expect that the problem is memory fragmentation. We are doing some work on that so we'll see what happens

Reply:
(also to be clear i'm not expecting any change in this PR, just making a note for the future)